spec/features/user_can_sign_up_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "rails_helper"
+
+RSpec.feature "SignUp", type: :feature do
+  scenario "Can sign up and details are saved" do
+    visit '/'
+    within '.header' do
+      click_link "Sign Up"
+    end
+    expect(page).to have_button("Sign Up")
+    fill_in "Username", with: "Username"
+    fill_in "Email", with: "user@example.com"
+    fill_in "Password", with: "Password"
+    fill_in "user_password_confirmation", with: "Password"
+    click_button "Sign Up"
+    expect(current_path).to eq "/posts"
+
+    user = User.find_by(username: "Username")
+
+    expect(user.username).to eq "Username"
+    expect(user.email).to eq "user@example.com"
+    expect(user.password).not_to eq "Password"
+  end
+
+  scenario "password is hidden" do
+    visit '/'
+    within '.header' do
+      click_link "Sign Up"
+    end
+    fill_in "Username", with: "Username"
+    fill_in "Email", with: "user@example.com"
+    fill_in "Password", with: "hidden"
+    expect(page).to_not have_content("hidden")
```

Comment:
given that you are staying on the same page, I think it would suffice to have "expect(page).to_not have_content("hidden")" only one time

Reply:
Another great spot, thank you. A bit of tired copy and pasting there I think!